### PR TITLE
Fix map-matching issue for segments at grid boundaries

### PR DESCRIPTION
**Pull request by smWork**
- Fixed number of rows and columns in grid
  - Fixed handling of row / column overflow

### CODE REVIEW

File: valhalla/thor/service.h [hunk 1]
```diff
@@ -130,6 +129,7 @@ class thor_worker_t {
   SOURCE_TO_TARGET_ALGORITHM source_to_target_algorithm;
   boost::optional<int> date_time_type;
   valhalla::meili::MapMatcherFactory matcher_factory;
+  valhalla::baldr::GraphReader& reader;
```
Comment (smWork) [inline valhalla/thor/service.h]: 'reader' should be declared after matcher_factory since it gets initialized from matcher_factory.

Comment (kevinkreiser) [inline valhalla/thor/service.h]: this was a good catch

**Comment by kevinkreiser** [comment]
@ptpt what do you think about the grid changes?

File: valhalla/meili/grid_range_query.h [hunk 2]
```diff
@@ -25,8 +25,8 @@ class GridRangeQuery
       bbox_(bbox),
       square_width_(square_width),
       square_height_(square_height),
-      ncols_((bbox.maxx() - bbox.minx()) / square_width),
-      nrows_((bbox.maxy() - bbox.miny()) / square_height),
+      ncols_(ceil((bbox.maxx() - bbox.minx()) / square_width)),
```
Comment (ptpt) [inline valhalla/meili/grid_range_query.h]: Shouldn't it be `floor`?

Comment (ptpt) [inline valhalla/meili/grid_range_query.h]: You are right; it should be `ceil` here. Type too fast.

**Comment by ptpt** [comment]
Would be nice to have a simple test at https://github.com/valhalla/valhalla/blob/master/test/grid_traversal.cc

**Pull request merged**